src/lib.rs

```diff
@@ -189,6 +191,58 @@ impl Tokenizer {
         })
     }
 
+    /// Tokenize a batch of multiple input strings.
+    ///
+    /// Each given input string is tokenized and the numeric representation
+    /// written to a row in the resulting two-dimensional matrix of shape
+    /// `(texts.len(), context_length)`, with the special `<start_of_text>`
+    /// token prepended, and `<end_of_text>` appended to each text.
+    ///
+    /// The individual input strings are lowercased before being tokenized, but
+    /// otherwise no pre-processing is performed.
+    ///
+    /// `context_length` is the maximum number of tokens per each text and
+    /// should be `77` for all current CLIP models. If tokenizing an input text
+    /// results in too many tokens, the token sequence will be truncated to fit
+    /// within the resulting row of length `context_length`, always including
+    /// the `<start_of_text>` and `<end_of_text>` marker tokens.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ndarray::array;
+    /// # use instant_clip_tokenizer::{Token, Tokenizer};
+    /// let tokenizer = Tokenizer::new();
+    /// let encoded = tokenizer.tokenize_batch(["Hi", "How are you?"], 5);
+    /// assert_eq!(encoded, array![
+    ///     [49406, 1883, 49407, 0, 0],
+    ///     [49406, 829, 631, 592, 49407],
+    /// ]);
+    /// ```
+    pub fn tokenize_batch<'a, I>(&self, texts: I, context_length: usize) -> Array2<u16>
+    where
+        I: IntoIterator<Item = &'a str>,
+        I::IntoIter: ExactSizeIterator,
+    {
+        if context_length < 3 {
+            panic!("context length must be at least 3");
+        }
+        let texts = texts.into_iter();
+        let mut result = Array2::zeros((texts.len(), context_length));
```

Comment:
If we're only using ndarray for `Array2` and the ability to iterate over it, it feels like we can make a minimal take on that without pulling in the large dependency?

Alternatively we should at least gate this with a `feature` gate and make the ndarray dependency optional.

Reply:
We need `Array2` as the return type. We could of course return a `Vec<u16>` instead with just the data, but that would put the burden on the caller to construct some matrix type from that (be it `ndarray::Array2` or `nalgebra::Matrix`).
For instance, the caller would need to think about the shape of the matrix, and whether the data is row-major or column-major order. Hence, returning a specific type representing a matrix is a lot more convenient and less error-prone for the caller.
We could use a different matrix type (e.g. from the `nalgebra` crate, or even roll our own with conversions to/from `ndarray`, ...) but `ndarray` seems what basically everyone is using in the context of ML so I chose that.

In summary, I would prefer to keep it that way, gated by a crate feature.

Reply:
Fair enough.